xla/service/host_offload_utils.cc

```diff
@@ -294,5 +296,69 @@ void SetHostComputeFrontendAttribute(HloInstruction& host_instruction) {
   host_instruction.set_frontend_attributes(frontend_attributes);
 }
 
+bool IsMoveToHostWithDynamicUpdateSlice(const HloInstruction* instr) {
+  if (!instr->IsCustomCall(kMoveToHostCustomCallTarget)) {
+    return false;
+  }
+
+  std::vector<const HloInstruction*> to_check = {instr};
+  absl::btree_set<const HloInstruction*> visited;
+
+  while (!to_check.empty()) {
+    const HloInstruction* current = to_check.back();
+    to_check.pop_back();
+
+    auto [_, inserted] = visited.insert(current);
+    if (inserted) {
+      if (current->user_count() == 0) {
```

Comment:
This check is not necessary, as the loop below will have 0 iteration in such a case.